### ISSUE TITLE
Dataclass InitVar's are not propagated to subclassed __post_init__'s

### ISSUE THREAD

**Issue by jgrey4296** [microsoft/pyright#3169]
**Describe the bug**
If a superclass dataclass has an initvar, Pyright gives an incorrect parameter count complaint in the subclass' `__post_init__`.
The code runs correctly in the python runtime.
This is also an error in Mypy.

Adding a duplicate InitVar to the subclass clears the complaint.

I'd guess it comes from dataclass.fields function, which doesn't return pseudo-fields?

**To Reproduce**
Shown below in the Code section. Create two dataclasses, with an initvar in the superclass, then try to implement a `__post_init__` in the subclass.

**Expected behavior**
Pyright should detect superclass InitVar's if the python runtime generates an `__init__` and  `__post_init__` with them.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from dataclasses import InitVar, dataclass, field

@dataclass
class MyData1:
    _name : InitVar[str] = field()
    name  : str              = field(init=False)

    def __post_init__(self, _name:str):
        self.name = f"blah {_name}"


@dataclass
class MyData2(MyData1):
    # _name : InitVar[str] = field() # uncomment this line to remove pyright error
    _age : InitVar[int] = field()
    age  : int              = field(init=False)

    def __post_init__(self, _name:str, _age:int):
        self.name = f"bloo {_name}"
        self.age = _age + 10


val   = MyData1("blah")
val2 = MyData2("bloo", 23)

assert(val.name == "blah blah")
assert(val2.name == "bloo bloo")
assert(val2.age == 33)

print(val)
print(val2)
```

**VS Code extension or command-line**
Command line 1.2.226 from Conda.
Python 3.10.0

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.229, which I just published. It will also be included in the next release of pylance.
